raiden/messages.py

```diff
@@ -1416,6 +1416,217 @@ def from_dict(cls, data):
         return expired_lock
 
 
+class SignedBlindedBalanceProof(SignedMessage):
+    """Message sub-field `onchain_balance_proof` for `RequestMonitoring`.
+    """
+
+    def __init__(
+            self,
+            channel_identifier: typing.ChannelID,
+            token_network_address: typing.TokenNetworkAddress,
+            nonce: typing.Nonce,
+            additional_hash: typing.AdditionalHash,
+            chain_id: typing.ChainID,
+            signature: typing.Signature,
+            balance_hash: typing.BalanceHash,
+            non_closing_signature=b'',
```

Comment:
forgot the type here

Reply:
This name doesn't make sense to me. There is very few points in time were a node will exchange a balance proof and the channel is closed. Usually we don't know who is the closer or the non-closer address when a balance proof is exchanged, so this variable can't really have a values (as it's currently named).

Reply:
A second thing, either this is a partner signature, in which case we should have it and it should be provided in the constructor, or it's a local signature, and this should be remove from the constructor and initialized to None in the `__init__`

Reply:
We need this keyword argument for roundtrip serialization (`.from_dict(.to_dict())`). The name itself comes directly from MS spec https://raiden-network-specification.readthedocs.io/en/latest/monitoring_service.html#monitor-request

Reply:
opened a PR to the spec https://github.com/raiden-network/spec/pull/213